File: Minimum_Number_of_Operations_to_Make_Array_Continuous.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var minOperations = function(nums) {
    const length = nums.length;
    let minOperations = length;
    const uniqueNums = new Set(nums);
    const sortedUniqueNums = Array.from(uniqueNums).sort((a, b) => a - b);
    let right = 0;

    for (let left = 0; left < sortedUniqueNums.length; left++) {
        while (right < sortedUniqueNums.length && sortedUniqueNums[right] < sortedUniqueNums[left] + length) {
            right++;
        }

        minOperations = Math.min(minOperations, length - (right - left));
    }

    return minOperations;    
};

File: Minimum_Number_of_Operations_to_Make_Array_Continuous.cpp
class Solution {
public:
    int minOperations(vector<int>& nums) {
        int length = nums.size();
        int minOperations = length;
        sort(nums.begin(), nums.end());
        nums.erase(unique(nums.begin(), nums.end()), nums.end());
        int right = 0;

        for (int left = 0; left < nums.size(); left++) {
            while (right < nums.size() && nums[right] < nums[left] + length) {
                right++;
            }

            minOperations = min(minOperations, length - (right - left));
        }

        return minOperations;        
    }
};
